src/symbol/projection.js

```diff
@@ -183,39 +187,41 @@ function getGlCoordMatrix(posMatrix: Float32Array,
     }
 }
 
-function project(point: {x: number, y: number}, matrix: Mat4, elevation: number = 0): ProjectedSymbol {
-    const pos = [point.x, point.y, elevation, 1];
+function project(x: number, y: number, matrix: Mat4, elevation: number = 0): Vec4 {
+    const pos = [x, y, elevation, 1];
     if (elevation) {
         vec4.transformMat4(pos, pos, matrix);
     } else {
         xyTransformMat4(pos, pos, matrix);
     }
     const w = pos[3];
-    return {
-        point: [pos[0] / w, pos[1] / w, pos[2] / w],
-        signedDistanceFromCamera: w
-    };
+    pos[0] /= w;
+    pos[1] /= w;
+    pos[2] /= w;
+    return pos;
```

Comment:
Same thought as above, could that `w` division be done implicitly by calling `vec3.transformMat4` above? `xyTransformMat4` leaves it untouched (`1` value).

Reply:
Good point, will try simplifying here!